### PR TITLE
add Next web3js to front-end templates

### PR DESCRIPTION
**Pull request by luu-alex**
# What :computer: 
* Adds a next web3.js to front-end templates examples
* Second thing updated with this PR
* Third thing updated with this PR

# Why :hand:
* Reason why first thing was added to PR
* Reason why second thing was added to PR
* Reason why third thing was added to PR

# Evidence :camera:
Include screenshots, screen recordings, or `console` output here demonstrating that your changes work as intended

# Notes :memo:
* Any notes/thoughts that the reviewers should know prior to reviewing the code?

### CODE REVIEW

**Review by jennyg0 — COMMENTED** [review]
thank you for the pr, looks great! have a few minor comments

File: templates/react/next-web3js/src/components/Context.tsx [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    name: "zkSync Sepolia Testnet",
+    rpcUrl: "https://rpc.ankr.com/eth_sepolia",
+    blockExplorerUrl: "https://sepolia.etherscan.io"
+  }
+  const zkSyncGoerliTestnet: Chain = {
+    id: "280",
+    name: "zkSync Goerli Testnet",
+    rpcUrl: "https://testnet.era.zksync.dev",
+    blockExplorerUrl: "https://goerli.explorer.zksync.io"
+  }
```
Comment (jennyg0) [inline templates/react/next-web3js/src/components/Context.tsx:34]: goerli can be removed

File: templates/react/next-web3js/src/components/Balance.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
+'use client'
+
+import { useState, useEffect } from 'react';
+import { Web3 } from 'Web3';
```
Comment (jennyg0) [inline templates/react/next-web3js/src/components/Balance.tsx:4]: should this be `from 'web3'`, getting a casing error here

File: templates/react/next-web3js/src/components/Context.tsx [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+const zkSync: Chain = {
+    id: "324",
+    name: "zkSync",
+    rpcUrl: "https://mainnet.era.zksync.io",
+    blockExplorerUrl: "https://explorer.zksync.io"
+  }
+  const zkSyncSepoliaTestnet: Chain = {
+    id: "300",
+    name: "zkSync Sepolia Testnet",
```
Comment (jennyg0) [inline templates/react/next-web3js/src/components/Context.tsx:25]: ZKsync here too

File: templates/react/next-web3js/src/components/contracts.ts [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+		name: 'transferFrom',
+		outputs: [{ internalType: 'bool', name: '', type: 'bool' }],
+		stateMutability: 'nonpayable',
+		type: 'function',
+	},
+] as const;
+
+
+export const daiContractConfig = {
+  address: "0x6Ff473f001877D553833B6e312C89b3c8fACa7Ac", // zkSync Era Sepolia Testnet DAI token address
```
Comment (jennyg0) [inline templates/react/next-web3js/src/components/contracts.ts:127]: ZKsync

File: templates/react/next-web3js/src/pages/index.tsx [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+      </Head>
+      <main className={`${styles.main} ${inter.className}`}>
+        <div>
+          <Connect />
+          {account.isConnected &&
+            <>
+              <hr />
+              <h2>Network</h2>
+              <p>
+                <strong>Make sure to connect your wallet to zkSync Testnet for full functionality</strong>
```
Comment (jennyg0) [inline templates/react/next-web3js/src/pages/index.tsx:45]: ZKsync here and in the title tag

File: templates/react/next-web3js/README.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
+This is a [Next.js](https://nextjs.org/) project bootstrapped with [`create-next-app`](https://github.com/vercel/next.js/tree/canary/packages/create-next-app).
```
Comment (jennyg0) [inline templates/react/next-web3js/README.md:1]: nice to have more specific readme like [this one](https://github.com/matter-labs/zksync-frontend-templates/tree/main/templates/react/next-wagmi#readme)

**Comment by luu-alex** [comment]
Feedback has been addressed :) Thank you!

File: templates/react/next-web3js/src/components/contracts.ts [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+		name: 'transferFrom',
+		outputs: [{ internalType: 'bool', name: '', type: 'bool' }],
+		stateMutability: 'nonpayable',
+		type: 'function',
+	},
+] as const;
+
+
+export const daiContractConfig = {
+  address: "0x6Ff473f001877D553833B6e312C89b3c8fACa7Ac", // ZKsync Era Sepolia Testnet DAI token address
```
Comment (jennyg0) [inline templates/react/next-web3js/src/components/contracts.ts:127]: thank you for the updates! sorry missed that official address for the DAI token is `0x604F0416e788779edB06c1A74a75FAad38384C6E`

File: templates/react/next-web3js/src/components/Context.tsx [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+        });
+      } else {
+        disconnect();
+      }
+    }
+  
+    const onNetworkChange = async (chainId: string) => {
+      const chainIdStr = web3?.utils.hexToNumberString(chainId)
+      const currentChain = chains.find((chain: any) => chain.id === chainIdStr);
+      setNetwork(currentChain ?? { id: chainIdStr, name: null, rpcUrl: null, unsupported: true });
```
Comment (jennyg0) [inline templates/react/next-web3js/src/components/Context.tsx:84]: Getting a type error: Argument of type 'Chain | { id: string | undefined; name: null; rpcUrl: null; unsupported: true; }' is not assignable to parameter of type 'SetStateAction<Chain | null>'.

File: templates/react/next-web3js/src/components/Context.tsx [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+  
+    const connect = async () => {
+      if (!getEthereumContext()) throw new Error("No injected wallets found");
+      web3 = new Web3((window as any).ethereum);
+
+
+      
+
+      const zkSyncPlugin = new ZkSyncPlugin(getEthereumContext());
+      web3.registerPlugin(zkSyncPlugin);
```
Comment (jennyg0) [inline templates/react/next-web3js/src/components/Context.tsx:96]: Also seeing a type error: Argument of type 'ZkSyncPlugin' is not assignable to parameter of type 'Web3PluginBase<unknown>'.

Comment (luu-alex) [reply templates/react/next-web3js/src/components/Context.tsx:98]: This issue will be fixed when the next version of web3 ZkSyncPlugin is published

File: templates/react/next-web3js/src/components/contracts.ts (hunk 7 above)
Comment (luu-alex) [reply templates/react/next-web3js/src/components/contracts.ts]: Updated

**Review by jennyg0 — APPROVED** [review]

File: templates/react/next-web3js/src/components/Context.tsx (hunk 9 above)
Comment (jennyg0) [reply templates/react/next-web3js/src/components/Context.tsx:98]: awesome @danforbes @luu-alex ! thank you for all the updates. one last thing is the casing looks like it was changed to `ZKsyncPlugin` from ZkSyncPlugin for the import.

Comment (danforbes) [reply templates/react/next-web3js/src/components/Context.tsx:98]: Thank you for spotting that @jennyg0 🙏🏻 Should be taken care of now ✅

**Review by jennyg0 — APPROVED** [review]

File: templates/react/next-web3js/README.md [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+Open [http://localhost:3000](http://localhost:3000) with your browser to see the result.
+
+In this project template, You can connect your wallet to the ZKsync testnet and see examples of the usage of the Web3.js library on ZKsync Era.
+
+
+## Learn More
+
+To learn more about [ZKsync](https://zksync.io) or [Web3.js](https://web3js.org/), check out the following resources:
+
+- [ZKsync Documentation](https://era.zksync.io/docs/dev) – learn about ZKsync features and API.
```
Comment (itsacoyote) [inline templates/react/next-web3js/README.md:26]:
```suggestion
- [ZKsync Documentation](https://docs.zksync.io) – learn about ZKsync features and API.
```

**Review by itsacoyote — CHANGES_REQUESTED** [review]

**Review by itsacoyote — APPROVED** [review]

**Review by jennyg0 — APPROVED** [review]

**Pull request merged**